### ISSUE TITLE
Unwrap outputs wrong values

### ISSUE THREAD

**Issue by jeanlaroche** [serge-sans-paille/pythran#1672]
The following script outputs vastly different values in python and pythran. I haven't been able to simplify it further (for example, if I pre-compute the values passed to np.unwrap() things work fine).
```
# pythran export test()
import numpy as np

def expo(freq, amp, N, phase=0):
    y = amp * np.exp(2.0j * np.pi * freq * np.arange(0, N, dtype=float) + phase)
    return y


def Func1(x, a1, low):
    s1 = 0j
    N = len(x)
    for ii in range(0, N, 2):
        out1 = x[ii] * a1 + s1
        s1 = -a1 * out1 + x[ii]
        low[ii // 2] = out1

def test():
    bufSize = 100
    a1 = 0.589412
    x = expo(0.11 / 8, 1, bufSize)
    low = np.zeros((bufSize),dtype = complex)
    Func1(x, a1, low)
    A = np.unwrap(np.angle(low))
    print(A[53:60])
```

**Comment by jeanlaroche** [comment]
After looking a bit more closely at low, I can see that this is caused by low having a 0 value! Let me investigate this a bit more.

**Comment by jeanlaroche** [comment]
Here's a much simpler reproducer
```
# pythran export test()
import numpy as np

def test():
    np.random.seed(0)
    low = np.random.randn(60) + 1j*np.random.randn(60)
    A = np.angle(low)
    A[50:60]=0
    #print(A)
    B = np.unwrap(A)
    print(B[0:60])
```
I understand that the random generator may not yield the same values in python and pythran, but the output of this script gives me:
```[[  -0.239741    -2.01774    -1.34099  0.00528633    0.312137    0.928287     -1.6943    -2.87785    0.114313    0.999539    -1.35923  -0.0594944    -1.34013    -2.15668    0.969512    -1.00512    0.641501      3.4625     5.62746     10.2622     15.0142     30.5926     57.9541      108.74     210.333     419.182      831.93     1663.32     3320.65     6638.64     13273.9     26542.5     53084.1      106166      212329      424656      849308 1.69861e+06 3.39722e+06 6.79444e+06 1.35889e+07 2.71777e+07 5.43555e+07 1.08711e+08 2.17422e+08 4.34844e+08 8.69688e+08 1.73938e+09 3.47875e+09  6.9575e+09 -1.3493e+10 -1.3493e+10 -1.3493e+10 -1.3493e+10 -1.3493e+10 -1.3493e+10 -1.3493e+10 -1.3493e+10 -1.3493e+10 -1.3493e+10]]```
Which is totally off.

**Comment by jeanlaroche** [comment]
I found the bug in unwrap.hpp. I'll create a PR

**Comment by serge-sans-paille** [comment]
Closed by  #1673 , by @jeanlaroche himself!
